### PR TITLE
Add IAR MSP430 example

### PR DESCRIPTION
**Pull request by LinuxJedi**
A basic MSP430 example for IAR IDE.

### CODE REVIEW

File: IDE/IAR-MSP430/include.am [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# vim:ft=automake
```
Comment (dgarske) [inline IDE/IAR-MSP430/include.am:1]:
```
Your pull request adds new files, but does not add them to include.am!
missing files (1 total):
----> IDE/IAR-MSP430/include.am
```

File: IDE/IAR-MSP430/main.c [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+#include <wolfssl/wolfcrypt/chacha20_poly1305.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+#include <msp430f5359.h>
+
+#define ECC_256_BIT_FIELD 32 /* 256-bit curve field */
+#define ECC_448_BIT_FIELD 56 /* 448-bit curve field */
```
Comment (dgarske) [inline IDE/IAR-MSP430/main.c:39]: This `ECC_448_BIT_FIELD` appears unused? Consider removing the curve448.h

File: IDE/IAR-MSP430/main.c [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+
+#define WOLF_GEN_MEM         (2*1024)
+
+static byte gWolfMem[WOLF_GEN_MEM];
+static byte generatedCiphertext[1024];
```
Comment (dgarske) [inline IDE/IAR-MSP430/main.c:44]: Maybe consider making the 1024 a macro?

**Review by dgarske — CHANGES_REQUESTED** [review]

**Comment by LinuxJedi** [comment]
Thanks for the input. Everything should be addressed now and a few extra things in there too.

**Review by dgarske — APPROVED** [review]

**Pull request merged**